samples/sample_decode/src/pipeline_decode.cpp

```diff
@@ -1884,6 +1897,23 @@ mfxStatus CDecodingPipeline::RunDecoding()
                 bErrIncompatibleVideoParams = true;
                 // need to go to the buffering loop prior to reset procedure
                 pBitstream = NULL;
+                sts = MFX_ERR_NONE;
+                continue;
+            } else if (MFX_ERR_REALLOC_SURFACE == sts) {
+                mfxVideoParam param;
+                m_pmfxDEC->GetVideoParam(&param);
+
+                m_pCurrentFreeSurface->frame.Info.CropW = param.mfx.FrameInfo.CropW;
```

Comment:
I think the update of `m_pCurrentFreeSurface` info should be done _inside_ `ReallocSurface()` function.

Reply:
Sure. I will change it. Thanks.